src/mscorlib/src/System/Buffer.cs

```diff
@@ -515,6 +518,15 @@ internal unsafe static void Memmove(byte* dest, byte* src, nuint len)
 
             do
             {
+                // This loop looks very costly since there
+                // appear to be a bunch of temporary values
+                // being created with the adds, but the jit
+                // (for x86 anyways) will convert each of
+                // these to use memory addressing operands.
+                // So the only cost is a bit of code size,
+                // which is made up for by the fact that
+                // we save on writes to dest/src.
```

Comment:
The lines seem to be a little short, you should break at least only after 80 chars